File: java/src/main/java/io/github/athxx/ason/Ason.java
package io.github.athxx.ason;

/**
 * Main entry point for ASON library.
 */
public class Ason {
    
    /**
     * Parse an ASON string.
     */
    public static Value parse(String input) throws ParseException {
        return Parser.parse(input);
    }
    
    /**
     * Serialize a Value to ASON format (data only).
     */
    public static String serialize(Value value) {
        return Serializer.serialize(value);
    }
    
    /**
     * Serialize a Value to ASON format with schema.
     */
    public static String serializeWithSchema(Value value) {
        return Serializer.serializeWithSchema(value);
    }
}

File: kotlin/src/main/kotlin/io/github/athxx/ason/Ason.kt
package io.github.athxx.ason

/**
 * Main entry point for ASON library.
 */
object Ason {
    
    /**
     * Parse an ASON string.
     */
    fun parse(input: String): Value = Parser.parse(input)
    
    /**
     * Serialize a Value to ASON format (data only).
     */
    fun serialize(value: Value): String = Serializer.serialize(value)
    
    /**
     * Serialize a Value to ASON format with schema.
     */
    fun serializeWithSchema(value: Value): String = Serializer.serializeWithSchema(value)
}

